pkg/registry/resource/resourceclaim/strategy.go

```diff
@@ -182,9 +182,18 @@ func dropDisabledDRAAdminAccessFields(newClaim, oldClaim *resource.ResourceClaim
 		// No need to drop anything.
 		return
 	}
+	if draAdminAccessFeatureInUse(oldClaim) {
+		// If anything was set in the past, then fields must not get
+		// dropped on potentially unrelated updates and, for example,
+		// adding a status with AdminAccess=true is allowed. The
+		// scheduler typically doesn't do that (it also checks the
+		// feature gate and refuses to schedule), but the apiserver
+		// would allow it.
+		return
+	}
 
 	for i := range newClaim.Spec.Devices.Requests {
-		if newClaim.Spec.Devices.Requests[i].AdminAccess != nil && !draAdminAccessFeatureInUse(oldClaim) {
+		if newClaim.Spec.Devices.Requests[i].AdminAccess != nil {
```

Comment:
why bother checking?

Reply:
For no good reason... [removed](https://github.com/kubernetes/kubernetes/compare/8687cab5aa12baa4775e393a1df9addfb00991fe..d1f0d24ef927b0ff0813d10e5c0716d6caf214df).